ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/services/buildActivityData.service.ts

```diff
@@ -30,11 +31,28 @@ export class buildActivityDataService {
     }
 
     private buildNoAnswer(question: QuestionDefinition): QuestionTypeModel {
-      return {
+      const noAnswerObj: QuestionTypeModel = {
         type: '',
         stableId: question.stableId,
         question: question.questionText,
       };
+      if(question.questionType === QuestionTypeEnum.Matrix) {
+        noAnswerObj.type = QuestionTypeEnum.Matrix;
+        noAnswerObj.matrixAnswer = this.generateDefaultMatrixAnswers(question);
+      }
+      return noAnswerObj;
+    }
+
+    private generateDefaultMatrixAnswers(question: QuestionDefinition): MatrixAnswer[] {
+      const matrixAnswer: MatrixAnswer[] = [];
+      question.rows.forEach(row => {
+        const mAnswer: MatrixAnswer = {
+          verticalAnswer: row,
+          horizontalAnswer: null
+        };
+        matrixAnswer.push(mAnswer);
+      });
+      return matrixAnswer;
```

Comment:
**_#optional_**

The method can be simplified a bit like:

```
 private generateDefaultMatrixAnswers(question: QuestionDefinition): MatrixAnswer[] {
     return question.rows.map(row => ({
      verticalAnswer: row,
      horizontalAnswer: null
    }));     
}
```

Reply:
It is already so. I pushed change right after creating this PR. refresh and you will see too. Anyway, thanks